### PR TITLE
Credentials documentation

### LINKED ISSUE #80: How to add credentials to Jenkins
Create a guide to help the users add credentials to Jenkins

### PR DESCRIPTION
**Pull request by samuelbernardolip**
Close #80 

Credentials documentation

### CODE REVIEW

File: docs/credentials/config_file.rst [hunk 1]
```diff
@@ -0,0 +1,122 @@
+Configurations
+===============
+
+config.yml: configuring credentials
+-----------------------------------
+
+In Jenkins configurations section is explained how to add the required
+credentials. Here is explained how to use credentials with JePL itself. *config*
+section of *config.yml* supports the keyword credentials that allows to define
```
Comment (orviz) [inline docs/credentials/config_file.rst:9]: ``credentials`` keyword between accents (\`\`credentials\`\`)

File: docs/credentials/config_file.rst [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+- secret text
+- username and password
+- secret file
+- ssh username with private key
+- certificate
+
+The difference between them are the keywords for each one. The association is
+made by the credentials id from Jenkins, where the credential type is defined.
+The following table resume the required parameters for each case:
```
Comment (orviz) [inline docs/credentials/config_file.rst:20]: resumes

File: docs/credentials/config_file.rst [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
++-------------------------------+----------------------------------------------+
+| Credential Type               | Properties                                   |
++===============================+==============================================+
+| secret text                   | **id**: jenkins identifier                   |
```
Comment (orviz) [inline docs/credentials/config_file.rst:25]: Personally I prefer accents rather than bold font

File: docs/credentials/jenkins.rst [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+1. If required, ensure you are logged in to Jenkins (as a user with the Credentials > Create permission).
+
+2. From the Jenkins home page (i.e. the Dashboard of the Jenkins classic UI), click Credentials > System on the left.
+
+3. Under System, click the Global credentials (unrestricted) link to access this default domain.
+
+4. Click Add Credentials on the left.
+
+**Note**: If there are no credentials in this default domain, you could also click the add some credentials link (which is the same as clicking the Add Credentials link).
```
Comment (orviz) [inline docs/credentials/jenkins.rst:17]: Use  `.. note::`

File: docs/credentials/jenkins.rst [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+========================
+
+Adding new global credentials
+-----------------------------
+
+To add new global credentials to your Jenkins instance:
+
+1. If required, ensure you are logged in to Jenkins (as a user with the Credentials > Create permission).
+
+2. From the Jenkins home page (i.e. the Dashboard of the Jenkins classic UI), click Credentials > System on the left.
```
Comment (orviz) [inline docs/credentials/jenkins.rst:11]: Highlight the Jenkins' menu items (Credentials, System, Global credentaials, Add credentials). Either accents or italic (I guess)

File: docs/credentials/jenkins.rst [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+    - Secret text - copy the secret text and paste it into the Secret field.
+
+    - Username and password - specify the credential’s Username and Password in their respective fields.
+
+    - Secret file - click the Choose file button next to the File field to select the secret file to upload to Jenkins.
+
+    - SSH Username with private key - specify the credentials Username, Private Key and optional Passphrase into their respective fields.
+
+    **Note**: Choosing Enter directly allows you to copy the private key’s text and paste it into the resulting Key text box.
```
Comment (orviz) [inline docs/credentials/jenkins.rst:37]: Use  `.. note::`

File: docs/credentials/jenkins.rst [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+    - Certificate - specify the Certificate and optional Password. Choosing Upload PKCS#12 certificate allows you to upload the certificate as a file via the resulting Upload certificate button.
+
+8. In the ID field, specify a meaningful credential ID value - for example, jenkins-user-for-xyz-artifact-repository. You can use upper- or lower-case letters for the credential ID, as well as any valid separator character. However, for the benefit of all users on your Jenkins instance, it is best to use a single and consistent convention for specifying credential IDs.
+
+**Note**: This field is optional. If you do not specify its value, Jenkins assigns a globally unique ID (GUID) value for the credential ID. Bear in mind that once a credential ID is set, it can no longer be changed.
```
Comment (orviz) [inline docs/credentials/jenkins.rst:43]: Use  `.. note::`

File: docs/credentials/intro.rst [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+jenkins-pipeline-library supports the available credentials bindings
+from `Jenkins automation server <https://jenkins.eosc-synergy.eu/>`_. The
+credentials can be defined in config.yml configuration file and used for each stage.
+
+Credentials stored in Jenkins can be used:
+
+  - as global credentials using environment variables
+  - by any specific Pipeline project represented in config.yml or Jenkinsfile
+  - in docker configuration files (Dockerfile or docker-compose.yml)
+
```
[the remark this answers is not in the document]
Comment (davrodgon) [reply docs/credentials/intro.rst:16]: Yes, it shows as a bulleted list to me.

**Review by davrodgon — APPROVED** [review]

File: docs/credentials/config_file.rst [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+|                               |                                              |
+|                               | **password_var**: environment variable to    |
+|                               | receive the password                         |
++-------------------------------+----------------------------------------------+
+
+In the next configuration examples, the values there have the following meaning:
+
+- jenkins_credential_id: this is the id associated to the Jenkins credential
+- ENV_*: name of the environment variable to be set with the correspondent value defined in Jenkins credential
+
```
[the remark this answers is not in the document]
Comment (samuelbernardolip) [reply docs/credentials/config_file.rst:70]: @orviz as @davrodgon confirmed, the list render works for me.
I change the italics as you pointed out... finishing the commit.

**Review by orviz — APPROVED** [review]

**Pull request merged**